StoryMerge/StoryMerger.cs

```diff
@@ -0,0 +1,217 @@
+﻿using Newtonsoft.Json;
+using S2VX.Game.Story;
+using S2VX.Game.Story.Note;
+using System.Collections.Generic;
+using System.IO;
+
+namespace StoryMerge {
+    public class StoryMerger {
+        private string[] Inputs { get; set; }
+        private string Output { get; set; }
+        private List<S2VXStory> InputStories { get; set; } = new List<S2VXStory>();
+
+        public StoryMerger(string[] inputs, string output) {
+            Inputs = inputs;
+            Output = output;
+        }
+
+        public Result Merge() {
+            var result = ValidateParameters();
+            if (!result.IsSuccessful) {
+                return result;
+            }
+
+            result = LoadInputs();
+            if (!result.IsSuccessful) {
+                return result;
+            }
+
+            var story = new S2VXStory();
+            var notesResult = MergeNotes(story);
+            var commandsResult = MergeCommands(story);
+            story.Save(Output);
+
+            var messages = new[] {
+                $"Merged {Inputs.Length} stories into \"{Output}\"",
+                notesResult.Message,
+                commandsResult.Message
+            };
+            var message = string.Join("\n\n", messages);
+
+            return new Result {
+                IsSuccessful = true,
+                Message = message
+            };
+        }
+
+        public Result ValidateParameters() {
```

Comment:
Command line programs that I've used have always printed out the proper usage as if you typed in `--help` upon failure. Maybe we should add that here? Something like: `Usage: dotnet run -i input1.s2ry input2.s2ry -o output.s2ry`

Reply:
You can use `dotnet run -- --help` or `dotnet run -- -h` or `dotnet run -- -h` to get the help input. The help text and options are automatically provided by the command line library I'm using. The reason you need the extra -- in there is because dotnet run has its own CLI options. Adding the extra -- tells the dotnet run command to pass any further options into the command line program it's building and running.

More details: https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-run#options

Reply:
I mean it seems inconsistent that it'd print out the usage if you have unrecognized/too many arguments but not when you have missing/too few arguments. It would be beneficial to the user to have the usage for the latter case. Not a huge deal, just a nitpick

Reply:
Ah yeah, I get it now. I made a change to invoke the help command manually if merge was not successful.